src/web/extractors.rs

```diff
@@ -1474,18 +1475,23 @@ impl FromRequest for BatchRequestOpt {
                 None => None,
                 Some(ref batch) if batch == "" || TRUE_REGEX.is_match(&batch) => None,
                 Some(batch) => {
-                    let pool = extrude_db_pool(&req).await?;
-                    let db = pool.get().await?;
-                    if db.validate_batch_id(batch.clone()).is_err() {
-                        return Err(ValidationErrorKind::FromDetails(
-                            format!(r#"Invalid batch ID: "{}""#, batch),
-                            RequestErrorLocation::QueryString,
-                            Some("batch".to_owned()),
-                            Some(ftags),
-                        )
-                        .into());
-                    }
-                    Some(batch)
+                    let transaction_pool: DbTransactionPool =
+                        DbTransactionPool::extract(&req).await?;
+
+                    transaction_pool
+                        .transaction(|db| async move {
```

Comment:
This is just a nit, but the batch could be valid for the duration of this transaction, but then not valid for the transaction that the handler is running in. This is mostly pedantic, because it would never happen in real life unless there was an attack happening or something, but this transaction really should be the same transaction as the handler that needs this check. I'm going to open a new bug to move from_request for BatchRequestOpt into a normal function.

Reply:
https://github.com/mozilla-services/syncstorage-rs/issues/710